lib/rubocop/cop/minitest/multiple_assertions.rb

```diff
@@ -74,10 +74,18 @@ def assertions_count_based_on_type(node)
         end
 
         def assertions_count_in_assignment(node)
-          # checking the direct expression is handled by assertion_method?
-          return 0 unless node.expression.block_type? || node.expression.numblock_type?
+          if node.masgn_type?
+            rhs = node.children.last
+
+            return rhs.children.sum { |child| assertions_count_based_on_type(child) } if rhs.array_type?
+            return assertion_method?(rhs) ? 1 : 0 if rhs.send_type?
+
+            # play it safe and bail if we don't have any explicit handling for whatever
+            # the RHS type is, since at this point we're already probably dealing with
+            # a pretty exotic situation that's unlikely in the real world
+            return 0
+          end
 
-          # this will only trigger the branches for :block and :numblock type nodes
           assertions_count_based_on_type(node.expression)
         end
```

Comment:
Can you refactor this method to the below?

```ruby
         def assertions_count_in_assignment(node)
           return assertions_count_based_on_type(node.expression) unless node.masgn_type?

           rhs = node.children.last
           case rhs.type
           when :array
             rhs.children.sum { |child| assertions_count_based_on_type(child) }
           when :send
             assertion_method?(rhs) ? 1 : 0
           else
             # play it safe and bail if we don't have any explicit handling for whatever
             # the RHS type is, since at this point we're already probably dealing with
             # a pretty exotic situation that's unlikely in the real world
             0
           end
         end
```